packages/authenticator/amplify_authenticator/lib/src/widgets/form_fields/confirm_sign_in_form_field.dart

```diff
@@ -773,3 +823,53 @@ class _ConfirmSignInDateFieldState extends _ConfirmSignInFormFieldState<String>
     );
   }
 }
+
+class _ConfirmSignInTotpSetupField extends ConfirmSignInFormField<String> {
+  const _ConfirmSignInTotpSetupField({
+    super.key,
+    required super.field,
+  }) : super._();
+
+  @override
+  _ConfirmSignInTotpSetupFieldState createState() =>
+      _ConfirmSignInTotpSetupFieldState();
+}
+
+class _ConfirmSignInTotpSetupFieldState
+    extends _ConfirmSignInFormFieldState<String> with TotpSetupFields {
+  @override
+  Uri get totpUri {
+    final totpUri = state.totpSetupUri;
+
+    assert(
+      totpUri != null,
+      'Expected TOTP setup uri in state for current screen, instead got $totpUri',
+    );
+    return totpUri!;
+  }
+}
+
+class _ConfirmSignInTotpAppButtonField extends ConfirmSignInFormField<String> {
+  const _ConfirmSignInTotpAppButtonField({
+    super.key,
+    required super.field,
+  }) : super._();
+
+  @override
+  _ConfirmSignInTotpAppButtonFieldState createState() =>
+      _ConfirmSignInTotpAppButtonFieldState();
+}
+
+class _ConfirmSignInTotpAppButtonFieldState
+    extends _ConfirmSignInFormFieldState<String>
+    with OpenAuthenticationAppButton {
+  @override
+  Uri get totpUri {
```

Comment:
Same Q. Can you please make a note why `setupUri` would ever be `null` here though?